### PR TITLE
docs(north-star): chart the LLM/quantized-inference domain

### PR DESCRIPTION
**Pull request by javimosch**
Automated maintenance run by automaintainer.

**Focus:** Pick a north-star domain; make one big move toward it each run.

----
WORK ALREADY IN FLIGHT — do not overlap:
These automaintainer pull requests are already open and awaiting review. Do NOT re-implement, refactor, or restructure the files they touch — pick non-overlapping work, and never recreate a change an open PR already makes. If your objective unavoidably overlaps one of these, choose a different, complementary improvement instead.

- PR #444 (am/am-7b5889-ti1rk5): test(math): cover atan/atan2 builtins incl. all four quadrants
    touches: atan_builtins_test.go
- PR #443 (am/am-7b5889-ti1pfj): [needs work] docs(examples): Q8_0 quantized GEMV example + pinned-output test
    touches: examples/README.md, examples/complex/quant_matmul.mfl, quant_matmul_example_test.go
- PR #442 (am/am-7b5889-ti1ltg): test(kernels): Q8_0 GEMV/GEMM integration — dot_q8 fused == manual inner loop
    touches: SPEC.md, kernel_q8_gemv_test.go, lexer.go, lexer_underscore_test.go
- PR #441 (am/am-7b5889-ti1jvh): test(kernels): pin int8/Q8_0 dot & peek numeric contracts
    touches: assign_undefined_test.go, kernel_q8_edge_test.go, types.go
- PR #440 (am/am-7b5889-ti1i0s): fix(racecheck): propagate happens-before through pre-spawn callees
    touches: check.go, classify_test.go, racecheck.go, racecheck_test.go
- PR #438 (am/am-7b5889-ti1g73): fix(codegen): short-circuit && / || instead of evaluating both operands
    touches: CHANGELOG.md, codegen.go, shortcircuit_test.go
- PR #425 (am/am-7b5889-thrhng): test(parser): cover parseFuncLit/parseStructLit error branches
    touches: buildwasm_test.go, check_test.go

**Branch:** `am/am-7b5889-ti1ti5`

> ℹ️ **Possible mislabel** — a `docs`-titled PR also edits source files: `inference_kernels_example_test.go`. Consider splitting so each PR is one concern with an accurate title.

**Diff:**
```
docs/NORTH-STAR-INFERENCE.md           | 150 +++++++++++++++++++++++++++++++++
 examples/complex/inference_kernels.mfl |  25 ++++++
 inference_kernels_example_test.go      |  92 ++++++++++++++++++++
 3 files changed, 267 insertions(+)
```

### CODE REVIEW

**Comment by javimosch** [comment]
⚠️ **Automaintainer intent check: OFF-OBJECTIVE**

FAIL — the diff only adds a new documentation file (docs/NORTH-STAR-INFERENCE.md) describing a domain rather than making a concrete engineering "big move" toward it.

The verification gate passed, but this change may not address the stated objective. Review before merging.

**Comment by javimosch** [comment]
**Automaintainer Review Verdict:** APPROVED — The PR is a valid two-part 'big move': commit 1 establishes the inference north star doc (following the NORTH-STAR-{BACKEND,WEB,GAMEDEV}.md pattern), and commit 2 delivers a concrete pure-MFL Tier-2 example (rmsnorm/silu/gelu/softmax/argmax over fp32 buffers) with end-to-end pinned tests. Math for all reference values verified: silu(±2), gelu(±1), rmsnorm([1,2,3]), softmax([1,2,3]), and the numerical-stability softmax test with logit=50. No overlap with the 7 in-flight PRs. No existing source files modified. The javimosch 'OFF-OBJECTIVE' flag applied to commit 1 alone; commit 2 directly addresses it with working code and locked numeric contracts.

**Comment by javimosch** [comment]
Approved by automaintainer review team — ready for human merge
